### ISSUE TITLE
Snap generating config file fails because of missing directory.

### ISSUE THREAD

**Issue by neoaggelos** [TheThingsNetwork/lorawan-stack#2355]
#### Summary

`ttn-lw-cli use <host> --user`  fails when CLI is installed using snap.

#### Steps to Reproduce

1. `sudo snap install ttn-lw-stack`
2. `ttn-lw-stack.ttn-lw-cli use localhost --user`

#### What do you see now?

```
 $ ttn-lw-stack.ttn-lw-cli use localhost --user
error:cmd/ttn-lw-cli/commands:fail_write (failed to write `HOME/snap/ttn-lw-stack/181/.config/.ttn-lw-cli.yml`)
    file=HOME/snap/ttn-lw-stack/181/.config/.ttn-lw-cli.yml
--- open HOME/snap/ttn-lw-stack/181/.config/.ttn-lw-cli.yml: no such file or directory
```

This happens because there is no directory `.config` directory inside the snap folder. If you create the directory by hand then it works.

#### What do you want to see instead?

The command to succeed (somehow)

#### Environment

3.7.0

#### How do you propose to implement this?

Two ideas:

- Create the destination directory if it does not exist already (not in favor of this one)
- When running in snap, add an extra default config file path `$SNAP_USER_COMMON` (which would be `HOME/snap/ttn-lw-stack/common`). Also, write configuration file there instead of the user config dir, which changes in every revision.

#### Can you do this yourself and submit a Pull Request?

yes, but let's discuss what's best first @htdvisser, @rvolosatovs

**Comment by rvolosatovs** [comment]
I think there are 2 different issues here:
1. Creating the destination path to the config if it does not exist. I'm largely in favor of this. In fact it's extremely annoying to use CLI programs, which do not do this.
2. We should indeed write the config to a stable location, which would persist across versions - that's a higher priority than (1.), because if this does not happen, then the snap package is quite unusable as it is now IMO.

**Comment by neoaggelos** [comment]
Closed with #2361
